### ISSUE TITLE
Class: Kitchen::Transport::SshFailed

### ISSUE THREAD

**Issue by Karthick-Gopalakrishnan** [QT-DevOps/DevOpsIssues#407]
Hello Team,

Am trying to do kitchen converge in Azure.... Centos vm is created when it comes to deployment part chef is failing.. Attached logs for your reference. 

[default-CentOs.log](https://github.com/QT-DevOps/DevOpsIssues/files/3051507/default-CentOs.log)
[kitchen.log](https://github.com/QT-DevOps/DevOpsIssues/files/3051508/kitchen.log)

recipe
 
![image](https://user-images.githubusercontent.com/48745218/55683109-1d462180-594d-11e9-9955-a52750c4655b.png)

![image](https://user-images.githubusercontent.com/48745218/55683134-5bdbdc00-594d-11e9-85f4-686dfdb551d7.png)

**Comment by santoshkalingwar** [comment]
Please check your platform section you are provided vm name as trusty 

https://directdevops.blog/2019/04/03/test-kitchen-setup-with-azure/

**Comment by Karthick-Gopalakrishnan** [comment]
Santhosh,

Vm name can be anything , If there is an issue with platform then my VM shouldn't be deployed. 
In my case the VM is deployed and accessible it fails when executing the recipe

**Comment by santoshkalingwar** [comment]
Did you created credentials file , provide client I'd , subscription ID , client secret 
, Install azure cli ,

**Comment by Karthick-Gopalakrishnan** [comment]
yes, my Azure VM is deployed and it is working. I have issues with executing the recipe that were my log goes into error

**Comment by shaikkhajaibrahim** [comment]
Please upload your cookbook

**Comment by Karthick-Gopalakrishnan** [comment]
[file_chef.zip](https://github.com/QT-DevOps/DevOpsIssues/files/3059441/file_chef.zip)

PFA.. Cookbook

**Comment by Karthick-Gopalakrishnan** [comment]
Update on the issue :

I used same cookbook and modified the platform to Ubuntu Ting!! Cookbook worked (VM deployed, File got created) 

So issue is with only centos and Redhat and am still working on it

**Comment by ghost** [comment]
yes it works on ubuntu
I am also looking at this issue

**Comment by Karthick-Gopalakrishnan** [comment]
Update 2 : 

I tried using Vagrant driver with centos os (CentOS Linux release 7.6.1810) and cookbook worked. 

Azure Vm with centos is still not working

**Comment by ghost** [comment]
give ssh-key in the transport section

**Comment by Karthick-Gopalakrishnan** [comment]
Hello Naresh,

Azure uses direct username and password ? Do we have to give ssh-key ?

Did it work in your environment giving SSH-KEY ?

**Comment by ghost** [comment]
in Azure authenticate type  have both username and password & ssh-key please check once
https://docs.microsoft.com/en-us/azure/virtual-machines/linux/mac-create-ssh-keys

**Comment by santoshkalingwar** [comment]
If issue are resolved , please close the issue

**Comment by Karthick-Gopalakrishnan** [comment]
Hello Santosh,

It is not resolved yet, I will try that ssh transport option

**Comment by Karthick-Gopalakrishnan** [comment]
Hello Naresh,

As you suggested i enabled SSH Transport option in my Kitchen.yml file. Added SSH keys and manually check if it is working - it worked. 

Private key is under the cookbook file_chef 

![image](https://user-images.githubusercontent.com/48745218/56271080-abb66200-6108-11e9-9ab9-ab2d4ce2cb4f.png)

When i do kitchen converge i get below error
![image](https://user-images.githubusercontent.com/48745218/56271148-d0123e80-6108-11e9-8fff-465017a7a600.png)

[default-Centos.log](https://github.com/QT-DevOps/DevOpsIssues/files/3088882/default-Centos.log)

I have installed below as per the log still same issue. 

chef gem install rbnacl
chef gem install rbnacl-libsodium
chef gem install bcrypt_pbkdf
